standard/expressions.md

```diff
@@ -4831,6 +4831,20 @@ The type of the expression `a ?? b` depends on which implicit conversions are
 
 Otherwise, `a` and `b` are incompatible, and a compile-time error occurs.
 
+> *Example*:
+>
+> <!-- Example: {template:"standalone-console-without-using", name:"DeclarationExpressions1"} -->
+> ```csharp
+> T M<T>(T a, T b) => a ?? b;
+>
+> string s = M(null, "text");
+> int i = M(19, 23);
+> ```
+>
+> The parameters to `M` are unconstrained. Therefore, the type arguments can be reference types, or nullable value types, as shown in the first call to `M`. The type arguments can also be non-nullable value types, as shown in the second call to `M`. When the type argument is a non-nullable value type, the expression `a ?? b` is `a`.
```

Comment:
```suggestion
> The parameters to `M` are unconstrained. Therefore, the type argument can be a reference type, or nullable value type, as shown in the first call to `M`. The type argument can also be a non-nullable value type, as shown in the second call to `M`. When the type argument is a non-nullable value type, the expression `a ?? b` is `a`.
```

(Just because there's only a single type parameter, and therefore a single type argument.)

Reply:
For "the expression `a ?? b` is `a`" should we have "is evaluated equivalently to" or something similar?

Reply:
@jskeet – No, if `a` is a non-nullable value type then `a` is non-null (think of the test as `(object)a == null` which for an IL target is a `box` and test) and so `a` is the result – it is not different handling *per se*. What is needed is to revise the bulleted list to cover this, see my earlier comment.

Reply:
My point is that the "is" here is a bit ambiguous. Perhaps "the value of the expression `a ?? b` is the value of `a`" or something like that? Just "is" seems wrong to me.

Reply:
@jskeet – I see, your "is evaluated equivalently to" threw me and “is the value of” is a lot better. But that leaves us with the problem that this example is not supported by the text, we could say its aspirational! ;-)